### PR TITLE
Add RemoveIDFromScopes

### LINKED ISSUE #6340: Concurrent container auto removes result in orphaned ips / network info
I created 16 containers concurrently with `--rm` specified and named 2 - 17 via:
```
docker run -d --rm --name {#} ubuntu /bin/sh -c 'a=0; while [ $a -lt 75 ]; do echo "line $a"; a=`expr $a + 1`; sleep 2; done;'
```
This resulted in 16 containers:
```
docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED              STATUS              PORTS               NAMES
b07f1ab1d37a        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       5
bb7c09c14676        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       4
111fb3a28833        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       12
2c733c90a322        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       13
5062fadd9df3        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       15
15ba4f59d9ba        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       11
fee7c5d822b8        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       2
c03a83fb6a85        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       7
f81b635bceca        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       3
4c672443ec4d        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       6
6a1cb2bb5636        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       16
b0e0de57d268        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       10
6d929e8800ef        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       17
bf9edaf8f983        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       14
29ff1eb418ea        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       9
ebca47a44ccf        ubuntu              "/bin/sh -c a=0; w..."   About a minute ago   Up About a minute                       8
```
The command provided to the container will run for roughly 2.5 minutes, so the containers will all exit at the same time.

This resulted in orphaned containerVM IP and network information as seen via `docker inspect bridge`:
```
"111fb3a28833150872a76f2c8396903faf093aa782ca0c56c86d794b0d584121": {
    "Name": "12",
    "EndpointID": "111fb3a28833150872a76f2c8396903faf093aa782ca0c56c86d794b0d584121",
    "MacAddress": "",
    "IPv4Address": "172.16.0.15/16",
    "IPv6Address": ""
  },
  "15ba4f59d9ba930d072d2002e7aea624253b8c26a0f40c5d22df7a74969ba5a9": {
    "Name": "11",
    "EndpointID": "15ba4f59d9ba930d072d2002e7aea624253b8c26a0f40c5d22df7a74969ba5a9",
    "MacAddress": "",
    "IPv4Address": "172.16.0.16/16",
    "IPv6Address": ""
  },
  "29ff1eb418eab31c3b42808d5d02f36237a6a485e1689a081403bdf92aa54d20": {
    "Name": "9",
    "EndpointID": "29ff1eb418eab31c3b42808d5d02f36237a6a485e1689a081403bdf92aa54d20",
    "MacAddress": "",
    "IPv4Address": "172.16.0.12/16",
    "IPv6Address": ""
  },
  "2c733c90a32257112b0ce8adac48e51189f88546df992b7bacee372a6e7f1462": {
    "Name": "13",
    "EndpointID": "2c733c90a32257112b0ce8adac48e51189f88546df992b7bacee372a6e7f1462",
    "MacAddress": "",
    "IPv4Address": "172.16.0.10/16",
    "IPv6Address": ""
  },
  "5062fadd9df3c262eabbc9dc98f4084725ef7dfd20506b103feb7dd1b357995e": {
    "Name": "15",
    "EndpointID": "5062fadd9df3c262eabbc9dc98f4084725ef7dfd20506b103feb7dd1b357995e",
    "MacAddress": "",
    "IPv4Address": "172.16.0.4/16",
    "IPv6Address": ""
  },
  "6a1cb2bb56361a64a0dd04110f8b130ef5794cc7e4c3d10825dd8d6ba77f2011": {
    "Name": "16",
    "EndpointID": "6a1cb2bb56361a64a0dd04110f8b130ef5794cc7e4c3d10825dd8d6ba77f2011",
    "MacAddress": "",
    "IPv4Address": "172.16.0.6/16",
    "IPv6Address": ""
  },
  "6d929e8800ef746937f96fd79ad93e28e1676f916b5e875c27f632bed31e6564": {
    "Name": "17",
    "EndpointID": "6d929e8800ef746937f96fd79ad93e28e1676f916b5e875c27f632bed31e6564",
    "MacAddress": "",
    "IPv4Address": "172.16.0.8/16",
    "IPv6Address": ""
  },
  "b07f1ab1d37abe079a8f191851f7b311952a22b83d23e83ab97840b8d9c7ff7b": {
    "Name": "5",
    "EndpointID": "b07f1ab1d37abe079a8f191851f7b311952a22b83d23e83ab97840b8d9c7ff7b",
    "MacAddress": "",
    "IPv4Address": "172.16.0.11/16",
    "IPv6Address": ""
  },
  "bb7c09c146761948a2ae0c84bf4e72662fd15e2c6ef1f97fb65b0f24d7354d15": {
    "Name": "4",
    "EndpointID": "bb7c09c146761948a2ae0c84bf4e72662fd15e2c6ef1f97fb65b0f24d7354d15",
    "MacAddress": "",
    "IPv4Address": "172.16.0.13/16",
    "IPv6Address": ""
  },
  "f81b635bceca9f289069fe3aad1781261abba7e30d5c10490a051af4f3ac5d18": {
    "Name": "3",
    "EndpointID": "f81b635bceca9f289069fe3aad1781261abba7e30d5c10490a051af4f3ac5d18",
    "MacAddress": "",
    "IPv4Address": "172.16.0.3/16",
    "IPv6Address": ""
  }
```
Environment details:
vSphere 6.0 u3 (nimbus)
ESXi 6.0 u3
Storage NFS
vic 1.2.1 deployed to resource pool via `--use-rp`
Reproducibility: easy
 
[logs](https://github.com/vmware/vic/files/1304638/orphaned_network.zip)

### PR DESCRIPTION
**Pull request by caglar10ur**
If the handle is gone that means someone removed the VM,
so unregister its network settings.

Fixes #6340

### CODE REVIEW

File: lib/portlayer/network/context.go [hunk 1]
```diff
@@ -858,6 +858,94 @@ func (c *Context) container(h *exec.Handle) (*Container, error) {
...
+	c.Lock()
+	defer c.Unlock()
+
+	uuid := uid.Parse(id)
+	if uuid == uid.NilUID {
+		return nil, fmt.Errorf("invalid container id %s", id)
+	}
+
+	con, ok := c.containers[uuid.String()]
+	if !ok {
```
Comment (hickeng) [inline lib/portlayer/network/context.go]: while we shouldn't ever have a nil entry in the map, it's not valid if we do and will cause panics. Can we check for `con == nil`

File: lib/portlayer/network/context.go [hunk 2]
```diff
@@ -892,53 +979,11 @@ func (c *Context) UnbindContainer(h *exec.Handle) ([]*Endpoint, error) {
 		// clear out assigned ip
```
Comment (hickeng) [inline lib/portlayer/network/context.go]: Worth noting that this is the reason we cannot combine UnbindContainer and UnbindContainerById further

File: lib/portlayer/network/context.go [hunk 3]
```diff
@@ -858,6 +858,94 @@ func (c *Context) container(h *exec.Handle) (*Container, error) {
...
+	}
+	// long id
+	delete(c.containers, con.ID().String())
+	// short id
+	delete(c.containers, con.ID().Truncate().String())
+	// name
+	delete(c.containers, con.Name())
+}
+
+func (c *Context) UnbindContainerByID(id string) ([]*Endpoint, error) {
```
Comment (hickeng) [inline lib/portlayer/network/context.go]: Should we call this RemoveIDFromScopes or similar? This isn't really an Unbind as we're not updating the container, just the tracking data for it.
It's not really evident unless looking at the handleEvent code what the difference is and definitely not why.

Comment (caglar10ur) [reply lib/portlayer/network/context.go]: Done

File: lib/portlayer/network/context.go (hunk 1 above)
Comment (caglar10ur) [reply lib/portlayer/network/context.go]: Done

**Comment by caglar10ur** [comment]
@hickeng pushed a new commit hopefully addressing your comments

**Pull request merged**